### PR TITLE
Wagtail 4.2 upgrade

### PR DESCRIPTION
**Pull request by katdom13**
- Added Python 3.11 and Wagtail 4.2 to the testing matrix

### CODE REVIEW

**Comment by katdom13** [comment]
For the maintainers,
I removed conditional code for Wagtail versions prior to 4.1 according to [these set of guidelines](https://github.com/wagtail/wagtail/wiki/Python-Package-Maintenance-Guidelines#setuppy--setupcfg--pyprojecttoml).
Please let me know if this is alright with you.

Thank you!

File: tox.ini [hunk 1]
```diff
@@ -12,15 +11,14 @@ basepython =
...
 deps =
     django3.2: Django>=3.2,<4.0
     django4.0: Django>=4.0,<4.1
     django4.1: Django>=4.1,<4.2
-    wagtail2.15: wagtail>=2.15,<2.16  # LTS
-    wagtail2.16: wagtail>=2.16,<2.17
-    wagtail3.0: wagtail>=3.0,<3.1
-    wagtail4.0: wagtail>=4.0,<4.1
+    wagtail4.1: wagtail>=4.1,<4.2  # LTS
+    wagtail4.2: wagtail>=4.2rc1,<4.3
```
Comment (nickmoreton) [inline tox.ini:21]:
```suggestion
    wagtail4.2: wagtail>=4.2,<5.0
```

Comment (katdom13) [reply tox.ini]: Hi @nickmoreton ,
I applied the suggestion. Thank you!

**Review by nickmoreton — APPROVED** [review]
Thank you @katdom13 looks good to me.

**Comment by nickmoreton** [comment]
@katdom13 I've been doing some cleanup and you could close this PR as it's superseded by: https://github.com/labd/wagtail-2fa/pull/223

**Comment by nickmoreton** [comment]
Superseded by: https://github.com/labd/wagtail-2fa/pull/227

**Pull request closed without merging**